katas/pokerhands/israel/poker.js

```diff
@@ -0,0 +1,433 @@
+                yy = caracteristicHand[1].toString();
+            }
+
+            if (caracteristicHand[2] < 10) {
+                zz = '0' + caracteristicHand[2].toString();
+            } else {
+                zz = caracteristicHand[2].toString();
+            }
+
+            return parseInt(x + yy + zz);
+        }
+        caracteristicHand = this.is2Pair(hand);
+        if (caracteristicHand[0]) {
+            const x = '3';
+            let yy = '';
+            let zz = '';
+
+            if (caracteristicHand[1] < 10) {
+                yy = '0' + caracteristicHand[1].toString();
+            } else {
+                yy = caracteristicHand[1].toString();
+            }
+
+            if (caracteristicHand[2] < 10) {
+                zz = '0' + caracteristicHand[2].toString();
+            } else {
+                zz = caracteristicHand[2].toString();
+            }
+
+            return parseInt(x + yy + zz);
+        }
+        caracteristicHand = this.isTri(hand);
+        if (caracteristicHand[0]) {
+            const x = '4';
+            let yy = '';
+            let zz = '';
+
+
+            if (caracteristicHand[1] < 10) {
+                yy = '0' + caracteristicHand[1].toString();
+            } else {
+                yy = caracteristicHand[1].toString();
+            }
+
+            if (caracteristicHand[2] < 10) {
+                zz = '0' + caracteristicHand[2].toString();
+            } else {
+                zz = caracteristicHand[2].toString();
+            }
+
+            return parseInt(x + yy + zz);
+        }
+        caracteristicHand = this.isStraight(hand);
+        if (caracteristicHand[0]) {
+            const x = '5';
+            let yy = '';
+            let zz = '';
+
+            if (caracteristicHand[1] < 10) {
+                yy = '0' + caracteristicHand[1].toString();
+            } else {
+                yy = caracteristicHand[1].toString();
+            }
+
+            if (caracteristicHand[2] < 10) {
+                zz = '0' + caracteristicHand[2].toString();
+            } else {
+                zz = caracteristicHand[2].toString();
+            }
+
+            return parseInt(x + yy + zz);
+        }
+        caracteristicHand = this.isFlush(hand);
+        if (caracteristicHand[0]) {
+            const x = '6';
+            let yy = '';
+            let zz = '';
+
+            if (caracteristicHand[1] < 10) {
+                yy = '0' + caracteristicHand[1].toString();
+            } else {
+                yy = caracteristicHand[1].toString();
+            }
+
+            if (caracteristicHand[2] < 10) {
+                zz = '0' + caracteristicHand[2].toString();
+            } else {
+                zz = caracteristicHand[2].toString();
+            }
+
+            return parseInt(x + yy + zz);
+        }
+        caracteristicHand = this.isFull(hand);
+        if (caracteristicHand[0]) {
+            const x = '7';
+            let yy = '';
+            let zz = '';
```

Comment:
variable names `x`, `yy`, `zz` do not describe what values they hold or what is their intended use, try to avoid those type of variable names and use ones that explain better what they are

Reply:
you are right jose, thanks a lot for your feedback!